src/pages/dar_application/ResearcherInfo_new.js

```diff
@@ -174,8 +213,26 @@ export default function ResearcherInfo(props) {
             type: 'button', // default button element type inside a form is "submit".
             className: 'button button-white btn-xs',
             style: { marginTop: 25 },
-            onClick: () => {}
-          }, ['Add Collaborator'])
+            onClick: () => addNewCollaborator(),
+            onChange: ({key: name, value,}) => formFieldChange({name, value}),
+          }, [
+            span({}, ['Add Collaborator'])
+          ]),
+
+          collaboratorsState
+            .map((state, index) => {
+              if (isNil(state)) {
```

Comment:
There shouldn't be an instance where an `undefined` value exists in any of the staff/collaborator lists. If the remove functionality is working as expected, then the target element is removed altogether from the array, reducing the total number of collaborators by one. 

If the list is initially empty, clicking `Add Collaborator` should add an element to the list with the relevant fields initialized to default values.